toolchain/ltrace.py

```diff
@@ -0,0 +1,104 @@
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM.
+# Author: Jayarama<jsomalra@linux.vnet.ibm.com>
+
+# Based on code by
+#   Author: Abdul Haleem <abdhalee@linux.vnet.ibm.com>
+
+
+import os
+import fnmatch
+from avocado import Test
+from avocado import main
+from avocado.utils import build
+from avocado.utils import process
+from avocado.utils import git
+from avocado.utils import distro
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Ltrace(Test):
+
+    def setUp(self):
+
+        backend = SoftwareManager()
+        dist = distro.detect()
+
+        """
+        Install  all the required dependencies
+        Building source tarball requires packages specific to os
+        that needs to be installed, if not installed test will stop.
+        """
+        if not backend.check_installed("gcc") and not backend.install("gcc"):
+            self.error("gcc is needed for the test to be run")
+
+        pkgs = ['git', 'wget', 'rpm', 'autoconf', 'automake',
+                'dejagnu', 'binutils']
+
+        if dist.name == 'sles':
+            sles_deps = ['build', 'libdw-devel', 'libelf-devel',
+                         'elfutils', 'binutils-devel', 'libtool', 'gcc-c++']
+            pkgs += sles_deps
+
+        elif dist.name == ('redhat'):
+            rhel_deps = ['elfutils-devel', 'elfutils-libelf-devel',
+                         'elfutils-libelf', 'elfutils-libs', 'libtool-ltdl']
+            pkgs += rhel_deps
+
+        elif dist.name == ('fedora'):
+            fedora_deps = ['elfutils-devel', 'elfutils-libelf-devel',
+                           'elfutils-libelf', 'elfutils-libs', 'libtool-ltdl']
+            pkgs += fedora_deps
+
+        elif dist.name == 'ubuntu':
+            ubuntu_deps = ['elfutils', 'libelf-dev', 'libtool',
+                           'libelf1', 'librpmbuild3', 'binutils-dev']
+            pkgs += ubuntu_deps
+        else:
+            self.log.warn("Unsupported OS!")
+
+        for pkg in pkgs:
+            if backend.install(pkg):
+                self.log.debug("%s installed successfull" % pkg)
+            else:
+                self.error("Error: Check repository !")
+
+        # clone the  ltrace tests from the GIT repo
+        # Source: git clone git://git.debian.org/git/collab-maint/ltrace.git
+
+        git.get_repo('git://git.debian.org/git/collab-maint/ltrace.git',
+                     destination_dir=os.path.join(self.srcdir, 'ltrace'))
+
+        self.src_lt = os.path.join(self.srcdir, "ltrace")
+        os.chdir(self.src_lt)
+        process.run('./autogen.sh')
+        process.run('./configure')
+        build.make(self.src_lt)
+
+    def test(self):
+        build.make(self.src_lt, extra_args='check')
+        for root, dirnames, filenames in os.walk('.'):
+            for filename in fnmatch.filter(filenames, '*.log'):
+                filename = os.path.join(root, filename)
+                logfile = filename[:-4] + ".log"
```

Comment:
OK I don't understand the need to remove `.log` and then add it back... Without this line it should work the same, shouldn't it?